src/pages/ProfilesExplorerView/domain/variables/ProfileMetricVariable.tsx

```diff
@@ -103,6 +104,9 @@ export class ProfileMetricVariable extends QueryVariable {
   onSelect = (newValue: string) => {
     reportInteraction('g_pyroscope_app_profile_metric_selected');
 
+    if (!this.state.skipUrlSync) {
```

Comment:
Checking skipUrlSync because we use this variable with skipping in other place

Reply:
another variant would be

```ts
 if (this.state.skipUrlSync) {
    this.changeValueTo(newValue);
} else {
  locationService.partial({ [`var-{this.state.name}`]: newValue })
}
```

need to URL encode the value there. 

doing it this way would be compatible (not conflict) with any built-in browser history history support in UrlSyncManager / interfaces